lib/boavizta.ts

```diff
@@ -0,0 +1,283 @@
+        const staticParamsCast = staticParams;
+        if ('verbose' in staticParamsCast) {
+            this.verbose = staticParamsCast.verbose as boolean ?? false;
+            staticParamsCast.verbose = undefined;
+        }
+        if ('allocation' in staticParamsCast) {
+            const allocation = staticParamsCast.allocation as string ?? "TOTAL";
+            // allocation has to be either TOTAL or LINEAR
+            if (["TOTAL", "LINEAR"].includes(allocation)) {
+                this.allocation = allocation;
+            } else {
+                throw new Error("Improper configure: Invalid allocation parameter. Either TOTAL or LINEAR");
+            }
+            staticParamsCast.allocation = undefined;
+        }
+        // if no valid provider found, throw error
+        this.validateProvider(staticParamsCast);
+        // if no valid instance_type found, throw error
+        await this.validateInstanceType(staticParamsCast);
+        // if no valid location found, throw error
+        await this.validateLocation(staticParamsCast);
+        this.sharedParams = Object.assign({}, staticParams);
+        return this.sharedParams
+    }
+
+    async validateLocation(staticParamsCast: object) {
+        if ('location' in staticParamsCast) {
+            const location = staticParamsCast.location as string ?? "USA";
+            const countries = await this.supportedLocations();
+            if (!(countries.includes(location))) {
+                throw new Error("Improper configure: Invalid location parameter: '" + location + "'. Valid values are : " + countries.join(", "));
+            }
+        }
+    }
+
+    async validateInstanceType(staticParamsCast: object) {
+        if (!('instance_type' in staticParamsCast)) {
+            throw new Error("Improper configure: Missing instance_type parameter");
+        }
+        if (!('provider' in staticParamsCast)) {
+            throw new Error("Improper configure: Missing provider parameter");
+        }
+        let provider = staticParamsCast.provider as string;
+        if (this.instanceTypes[provider] === undefined || this.instanceTypes[provider].length === 0) {
+            this.instanceTypes[provider] = await this.supportedInstancesList(provider)
+        }
+        if ('instance_type' in staticParamsCast) {
+            if (!(this.instanceTypes[provider].includes(staticParamsCast.instance_type as string))) {
+                throw new Error("Improper configure: Invalid instance_type parameter: '" + staticParamsCast.instance_type + "'. Valid values are : " + this.instanceTypes[provider].join(", "));
+            }
+        } else {
+            throw new Error("Improper configure: Missing instance_type parameter");
+        }
+    }
+
+    validateProvider(staticParamsCast: object) {
+        if (!('provider' in staticParamsCast)) {
+            throw new Error("Improper configure: Missing provider parameter");
+        } else {
+            const supportedProviders = ["aws"];
```

Comment:
There is a Boavizta endpoint to return supported providers

Reply:
Updated.